api/tests/src/db/models/factories.py

```diff
@@ -1907,4 +1913,16 @@ class Meta:
     opportunity = factory.SubFactory(OpportunityFactory)
     opportunity_id = factory.LazyAttribute(lambda o: o.opportunity.opportunity_id)
 
-    opportunity_data = factory.LazyAttribute(lambda o: o.opportunity_data)
+    opportunity_data = factory.LazyAttribute(lambda o: SCHEMA.dump(o.opportunity))
+
+
+class JobLogFactory(BaseFactory):
+    class Meta:
+        model = task_models.JobLog
+
+    job_id = Generators.UuidObj
+    job_type = factory.LazyAttribute(
+        lambda _: random.choice(["StoreOpportunityVersionTask", "SetCurrentOpportunitiesTask"])
+    )
+    job_status = factory.Iterator(JobStatus)
```

Comment:
I'd probably say to just make this always the success value since most tests would only care about that (or want to override it to started/failed).